conformance/fhir-ig-us-core/src/test/java/com/ibm/fhir/ig/us/core/test/v400/ConformanceTest.java

```diff
@@ -507,4 +518,89 @@ public void testUSCoreRaceExtension4() throws Exception {
+                        .coding(Coding.builder()
+                            .system(Uri.of("http://hl7.org/fhir/us/core/STU4/CodeSystem-condition-category.html"))
+                            .code(Code.of("health-concern"))
+                            .build())
+                        .build())
+                .build();
+
+        FHIRValidator validator = FHIRValidator.validator();
+        List<Issue> issues = validator.validate(condition, "http://hl7.org/fhir/us/core/StructureDefinition/us-core-condition");
+
+        issues.forEach(System.out::println);
+
+        Assert.assertEquals(countErrors(issues), 0);
+        // warning 1:  us-core-1: A code in Condition.category SHOULD be from US Core Condition Category Codes value set.
+        // causedBy
+        // warning 2:  Membership check was not performed: value set 'http://hl7.org/fhir/us/core/ValueSet/us-core-condition-code' is empty or could not be expanded
+        Assert.assertEquals(countWarnings(issues), 2);
+    }
+
+    @Test
+    void testProvenanceWithMultipleAgent() throws FHIRValidationException, FHIRPathException, FHIRPatchException {
+        Provenance provenance = Provenance.builder()
+                .text(Narrative.EMPTY)
+                .target(Reference.builder()
+                        .reference("Observation/123")
+                        .build())
+                .recorded(Instant.now())
+                .agent(Agent.builder()
+                        .who(Reference.builder()
+                            .reference("Patient/123")
+                            .build())
+                        .build())
+                .agent(Agent.builder()
+                        .who(Reference.builder()
+                            .reference("Practitioner/abc")
+                            .build())
+                        .build())
+                .agent(Agent.builder()
+                        .who(Reference.builder()
+                            .reference("Device/xyz")
+                            .build())
+                        .onBehalfOf(Reference.builder()
+                            .reference("RelatedPerson/bleh")
+                            .build())
+                        .build())
+                .build();
+
+        FHIRValidator validator = FHIRValidator.validator();
+        List<Issue> issues = validator.validate(provenance, "http://hl7.org/fhir/us/core/StructureDefinition/us-core-provenance");
+
+        issues.forEach(System.out::println);
+
+        Assert.assertEquals(countErrors(issues), 1);
+        Assert.assertEquals(countWarnings(issues), 0);
+
+        Reference refToAdd = Reference.builder()
+            .reference("Organization/fix")
+            .build();
+        provenance = FHIRPathUtil.add(provenance, "Provenance.agent[1]", "onBehalfOf", refToAdd);
+        issues = validator.validate(provenance, "http://hl7.org/fhir/us/core/StructureDefinition/us-core-provenance");
+        Assert.assertEquals(countErrors(issues), 0);
```

Comment:
Perhaps check countWarnings after this update, too.  The error is gone, which is what mattered, so maybe not worth it.

Reply:
yeah, i was mainly just interested in showing that the error is gone now that the required field was added